File: app/src/main/java/com/kota/Bahamut/DataModels/ArticleTemp.java
package com.kota.Bahamut.DataModels;

import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.StreamCorruptedException;
import org.json.JSONException;
import org.json.JSONObject;

public class ArticleTemp {
    public static final int version = 1;
    public String content = "";
    public String header = "";
    public String title = "";

    public JSONObject exportToJSON() throws JSONException {
        JSONObject obj = new JSONObject();
        obj.put("header", this.header);
        obj.put("title", this.title);
        obj.put("content", this.content);
        return obj;
    }

    public void importFromJSON(JSONObject obj) throws JSONException {
        this.header = obj.getString("header");
        this.title = obj.getString("title");
        this.content = obj.getString("content");
    }

    public void exportToStream(ObjectOutputStream aStream) throws IOException {
        aStream.write(1);
        aStream.writeUTF(this.header);
        aStream.writeUTF(this.title);
        aStream.writeUTF(this.content);
    }

    public void importFromStream(ObjectInputStream aStream) throws IOException {
        aStream.readInt();
        this.header = aStream.readUTF();
        this.title = aStream.readUTF();
        this.content = aStream.readUTF();
    }
}

File: app/src/main/java/com/kota/Bahamut/DataModels/ArticleTemp.kt
package com.kota.Bahamut.DataModels

import org.json.JSONException
import org.json.JSONObject
import java.io.IOException
import java.io.ObjectInputStream
import java.io.ObjectOutputStream

class ArticleTemp {
    var content: String = ""
    var header: String = ""
    var title: String = ""

    @Throws(JSONException::class)
    fun exportToJSON(): JSONObject {
        val obj = JSONObject()
        obj.put("header", header)
        obj.put("title", title)
        obj.put("content", content)
        return obj
    }

    @Throws(JSONException::class)
    fun importFromJSON(obj: JSONObject) {
        header = obj.getString("header")
        title = obj.getString("title")
        content = obj.getString("content")
    }

    @Throws(IOException::class)
    fun exportToStream(aStream: ObjectOutputStream) {
        aStream.write(1)
        aStream.writeUTF(header)
        aStream.writeUTF(title)
        aStream.writeUTF(content)
    }

    @Throws(IOException::class)
    fun importFromStream(aStream: ObjectInputStream) {
        aStream.readInt()
        header = aStream.readUTF()
        title = aStream.readUTF()
        content = aStream.readUTF()
    }

    companion object {
        const val version = 1
    }
}
